lte/gateway/c/session_manager/SessionStateEnforcer.cpp

```diff
@@ -106,57 +114,180 @@ bool SessionStateEnforcer::m5g_init_session_credit(
   return true;
 }
 
-void SessionStateEnforcer::handle_session_init_rule_updates(
+bool SessionStateEnforcer::m5g_update_session_context(
     SessionMap& session_map, const std::string& imsi,
-    SessionState& session_state) {
-  uint32_t l_teid = 0;
-  auto itp        = pdr_map_.equal_range(imsi);
+    std::unique_ptr<SessionState>& session_state,
+    SessionUpdate& session_update) {
+  uint32_t gnode_teid = 0;
+  bool gnb_teid_get   = true;
+
+  /* Check and update latest session rules
+   * we got gnodeb TEID, and IP address details here
+   */
+  gnode_teid = update_session_rules(imsi, session_state, gnb_teid_get);
+  if (!gnode_teid) return false;
+  auto session_id                        = session_state->get_session_id();
+  SessionStateUpdateCriteria& session_uc = session_update[imsi][session_id];
+  uint32_t cur_version                   = session_state->get_current_version();
+  session_state->set_fsm_state(CREATED, session_uc);
+  session_state->set_current_version(++cur_version, session_uc);
+  /* Reset the upf resend throttle counter, send the session creation
+   * request to UPF
+   */
+  session_state->reset_throttle();

```

Comment:
is this retransmission counter? if so please rename throttle to rtx_counter

Reply:
Fixed, resolved.